codec/src/main/java/io/netty/handler/codec/digests/MD5Digest.java

```diff
@@ -0,0 +1,239 @@
+package io.netty.handler.codec.digests;
+
+/**
+ * implementation of MD5 as outlined in "Handbook of Applied Cryptography",
+ * pages 346 - 347.
+ */
+public class MD5Digest extends GeneralDigest {
+  public static final int DIGEST_LENGTH = 16;
+
+  private int H1, H2, H3, H4; // IV's
+
+  private int[] X = new int[16];
+  private int xOff;
+
+  /**
+   * Standard constructor
+   */
+  public MD5Digest() {
+    reset();
+  }
+
+  public int getDigestSize() {
+    return DIGEST_LENGTH;
+  }
+
+  protected void processWord(byte[] in, int inOff) {
```

Comment:
I Agree. I just wanted to offer the alternatives, because the main motivation was initialisation and multithreading concerns of JDK digest. I would go ahead with ByteBuf. Any more opinions @normanmaurer / @trustin ?